frontend/src/stores/userStore.ts

```diff
@@ -0,0 +1,94 @@
+import { ValidRelationships, type Relationship } from '@/types/Relationship';
+import { getRequest, patchRequest, postRequest } from '@/utils/apiRequests';
+import { defineStore } from 'pinia';
+import type { User } from "../types/User";
+
+export const useUserStore = defineStore("users", {
+
+  //  actions == data definitions
+  state: () => ({
+      allUsers: <User[]>[],
+      me: <User>{},
+  }),
+  // getters == computed values
+  getters: {
+    getAllUsers: (state) => state.allUsers,
+    getMe: (state) => state.me
+  },
+  // actions == methods
+  actions: {
+    async login() {
+      const loginData = {
+        name: "renoster",
+          password: 'R'
+        }
+        await postRequest("login", loginData);
+    },
+
+    async refreshMe() {
+      try {
+        const data = await getRequest("me");
+        this.me = data.data;
+      }
+      catch (e) {
+        console.error(e);
+      }
+    },
+
+    async refreshAllUsers() {
+      try {
+        const data = await getRequest("users");
+        this.allUsers = data.data;
+      }
+      catch (e) {
+        console.error(e);
+      }
+    },
+
+    async getRelationship(source: number, target: number)
+    {
+      return await (await getRequest(`user-relationships/${source}/${target}`)).data
+    },
```

Comment:
does this endpoint exist in this format? I only see a `/:id` endoint, no `/:source/:target`

Reply:
re-added endpoint with logic in .service